sherpa/optmethods/optfcts.py

```diff
@@ -445,9 +502,32 @@ def stat_cb0(x_new):
 #
 # Monte Carlo
 #
-def montecarlo(fcn, x0, xmin, xmax, ftol=EPSILON, maxfev=None, verbose=0,
-               seed=74815, population_size=None, xprob=0.9,
-               weighting_factor=0.8, numcores=1, rng=None
+def random_start(rng: random.RandomType | None,
+                 xmin: np.ndarray,
+                 xmax: np.ndarray
+                 ) -> np.ndarray:
+    """Create an array of values between xmin and xmax."""
+
+    out = np.zeros_like(xmin)
+    for idx, (minval, maxval) in enumerate(zip(xmin, xmax)):
+        out[idx] = random.uniform(rng, minval, maxval)
+
+    return out
```

Comment:
This function is unnecessary. Numpy itself allows `low` and `high` argument in the random number generators to be arrays already. Remove this function and inline the lumpy call directly where needed:
```Python
rand_start = random.uniform(rng, xmin, xmax)
```

Performance will be better (no Python loop), but more importantly, the code it easier to understand because it avoids an extra level of indirection.

Reply:
I know this function was there before, but sine this commit touches it, it might as well remove it altogether. I checked the docs back to lumpy 1.20 and at least the far `low` and `high` can be arrays.

Reply:
Note that `random.uniform` is actually `sherpa.utils.random.uniform`. Now, we can do what we like with this routine, but it just needs some work to check.

For reference, this routine will take the `rng` argument and do something sensible if `rng` is `None`.

Reply:
It should just pass through the arrays, because `sherpa.utils.random.uniform` call numpy in both cases and will just do the right thing: https://github.com/sherpa/sherpa/blob/main/sherpa/utils/random.py#L188

Reply:
Which will means tat tis function is not needed, `just all sherpa.utils.random.uniform` directly.

Reply:
I've removed this routine (and added a test just to show that `sherpa.utils.random.uniform` is doing the right thing).

There's a 50/50 chance the tests will fail because of some other recent change to OTS packages ....

Reply:
@hamogu - is this change okay?